### ISSUE TITLE
Error handling for negative tests in hammer

### ISSUE THREAD

**Issue by sghai** [SatelliteQE/robottelo#3611]
This is the issue specifically for negative scenarios. Currently, we are checking the returncode however, we should actually check the expected error message too. This is required because in hammer if we pass `incorrect_id` of an entity as an option then some cmd fails like:

``` bash
]# hammer  -u admin -p changeme hostgroup create --name text --subnet-id 12345
Could not create the hostgroup:
  ERROR:  insert or update on table "hostgroups" violates foreign key constraint "hostgroups_subnet_id_fk"
  DETAIL:  Key (subnet_id)=(12345) is not present in table "subnets".

```

Since we are just validating `returncode` which is not sufficient like below where we are just checking returncode, because never know if cmd fails with expected error message or some db error

``` python
def test_negative_delete_by_id(self):
        """Create HostGroup then delete it by wrong ID

        @id: 047c9f1a-4dd6-4fdc-b7ed-37cc725c68d3

        @assert: HostGroup is not deleted
        """
        for entity_id in invalid_id_list():
            with self.subTest(entity_id):
                with self.assertRaises(CLIReturnCodeError):
                    HostGroup.delete({'id': entity_id})

```

This can be done like below:

``` python
        subnet_id = gen_string('numeric', 4)
        with self.assertRaises(CLIReturnCodeError) as exception:
            HostGroup.create({
                'name': gen_string('alpha'),
                'subnet-id': subnet_id
            })
        self.assertIs(
            exception.exception.stderr,
            'Could not find subnet {0}'.format(subnet_id)

```

**Comment by renzon** [comment]
Once it impacts almost all files from cli, I am gonna create issue for each file so PR are smaller and easier to review

**Comment by rplevka** [comment]
@renzon - just keep this issue as a tracker, create a list of files with checkboxes so we can keep track of those you've already addressed. e.g.:
- [ ] test_organization.py
- [ ] test_user.py

**Comment by elyezer** [comment]
One thing to have in mind is if we start to do localization testing this will all break since the messages are going to be translated. If localization tests is something to have in the future then this should be considered before starting working on this issue.

**Comment by pondrejk** [comment]
Adding a tracking checklist (for lack of a better place):

- [ ] test_abrt.py -- no negative cases
- [ ] test_activationkey.py -- #3716
- [ ] test_architecture.py -- #3747 
- [ ] test_bootstrap_script.py
- [ ] test_capsule.py
- [ ] test_capsule_installer.py
- [ ] test_classparameters.py
- [ ] test_computeresource.py
- [ ] test_contenthost.py
- [ ] test_contentview.py
- [ ] test_contentviewfilter.py
- [ ] test_discoveredhost.py
- [ ] test_docker.py
- [ ] test_domain.py
- [ ] test_environment.py
- [ ] test_errata.py
- [ ] test_fact.py
- [ ] test_filter.py
- [ ] test_globalparam.py
- [ ] test_gpgkey.py
- [ ] test_hammer.py
- [ ] test_host.py
- [ ] test_host_collection.py
- [ ] test_hostgroup.py
- [ ] test_import.py
- [ ] test_installer.py
- [ ] test_lifecycleenvironment.py
- [ ] test_location.py
- [ ] test_medium.py
- [ ] test_model.py
- [ ] test_myaccount.py
- [ ] test_operatingsystem.py
- [ ] test_organization.py
- [ ] test_partitiontable.py
- [ ] test_ping.py
- [ ] test_product.py
- [ ] test_puppetmodule.py
- [ ] test_report.py
- [ ] test_repository.py
- [ ] test_repository_set.py
- [ ] test_role.py
- [ ] test_satellitesync.py
- [ ] test_sso.py
- [ ] test_subnet.py
- [ ] test_subscription.py
- [ ] test_syncplan.py
- [ ] test_template.py -- no negative cases
- [ ] test_user.py -- #3745 
- [ ] test_usergroup.py -- #3743
- [ ] test_variables.py -- all stubbed

**Comment by rochacbruno** [comment]
Lets consider the #3749 before continue with this.

**Comment by rochacbruno** [comment]
What is the current status of this @renzon @pondrejk ?

**Comment by pondrejk** [comment]
Well, at some point in past we decided on a meeting to halt this activity, so I'm not sure if we want to proceed or put this to backlog. Maybe a question for a meeting.

**Comment by renzon** [comment]
@pondrejk +1

**Comment by rochacbruno** [comment]
@pondrejk @renzon 

This issue has been implemented on CLI side, but we need to file a new issue on nailgun to address API, this issue will be opened until we move the list of files to the new issue.

@svtkachenko

**Comment by ntkathole** [comment]
Bugzilla closed as Wontfix. Please feel free to re-open, if this still needs fixing.
